mlrun/model_monitoring/db/tsdb/timescaledb/queries/timescaledb_predictions_queries.py

```diff
@@ -0,0 +1,434 @@
+            )
+
+        return df
+
+    def get_avg_latency(
+        self,
+        endpoint_ids: Union[str, list[str]],
+        start: Optional[datetime] = None,
+        end: Optional[datetime] = None,
+        get_raw: bool = False,
+    ) -> Union[pd.DataFrame, list[v3io_frames.client.RawFrame]]:
+        """Get average latency with automatic pre-aggregate optimization, returning single value per endpoint."""
+
+        # Convert single endpoint to list for consistent handling
+        if isinstance(endpoint_ids, str):
+            endpoint_ids = [endpoint_ids]
+
+        # Set default start time and get end time
+        start = start or (mlrun.utils.datetime_now() - timedelta(hours=24))
+        # Prepare time range with auto-determined interval
+        start, end, interval = TimescaleDBQueryBuilder.prepare_time_range_and_interval(
+            self._pre_aggregate_handler, start, end
+        )
+
+        table_schema = self.tables[mm_schemas.TimescaleDBTables.PREDICTIONS]
+        filter_query = TimescaleDBQueryBuilder.build_endpoint_filter(endpoint_ids)
+
+        def build_pre_agg_query():
+            # Calculate overall average in SQL across all time buckets
+            # Use subquery to get time-bucketed data, then AVG over those results
+            subquery = table_schema._get_records_query(
+                start=start,
+                end=end,
+                columns_to_filter=[
+                    timescaledb_schema.TIME_BUCKET_COLUMN,
+                    mm_schemas.ModelEndpointSchema.AVG_LATENCY,
+                    mm_schemas.WriterEvent.ENDPOINT_ID,
+                ],
+                filter_query=filter_query,
+                agg_funcs=["avg"],
+                interval=interval,
+                use_pre_aggregates=True,
+            )
+
+            # Use helper to build endpoint aggregation query
+            return TimescaleDBQueryBuilder.build_endpoint_aggregation_query(
+                subquery=subquery,
+                aggregation_columns={
+                    mm_schemas.ModelEndpointSchema.AVG_LATENCY: f"AVG({mm_schemas.ModelEndpointSchema.AVG_LATENCY})"
+                },
+            )
+
+        def build_raw_query():
+            # Single aggregated value across entire time range
+            columns = [
+                f"{mm_schemas.WriterEvent.ENDPOINT_ID} AS {mm_schemas.WriterEvent.ENDPOINT_ID}",
+                f"AVG({mm_schemas.EventFieldType.LATENCY}) AS {mm_schemas.ModelEndpointSchema.AVG_LATENCY}",
+            ]
+            group_by_columns = [mm_schemas.WriterEvent.ENDPOINT_ID]
+
+            # Add additional filter to exclude invalid latency values
+            latency_col = mm_schemas.EventFieldType.LATENCY
+            latency_filter = f"{latency_col} IS NOT NULL AND {latency_col} > 0"
+            if filter_query:
+                enhanced_filter_query = f"{filter_query} AND {latency_filter}"
+            else:
+                enhanced_filter_query = latency_filter
```

Comment:
Maybe
```suggestion
            enhanced_filter_query = f"{filter_query} AND {latency_filter}" if filter_query else latency_filter
```
or use `TimescaleDBQueryBuilder.combine_filters() as done elsewhere`.

Reply:
will fix